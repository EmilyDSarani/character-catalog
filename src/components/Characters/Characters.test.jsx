import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router';
import Characters from './Characters'

it('should render characters', () => {
    const { container } = render(
    <MemoryRouter> 
    <Characters characters={[    
    {
    name: 'Sarani',
    id: 1,
    image:'https://place-puppy.com/'
}
    ]}/>
    </MemoryRouter> 
    );
    expect(container).toMatchSnapshot();
});

//Data Type- type of data
//primatives
//the strings, number, arrays, objects, booleans, undefined/null 

//when memoryrouter is not there, the error is "cant use link outside of a router"