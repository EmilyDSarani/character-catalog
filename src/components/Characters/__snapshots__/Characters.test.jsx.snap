// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render characters 1`] = `
<div>
  <ul
    aria-label="characters"
    class="list"
  >
    <li
      class="item"
    >
      <a
        href="/characters/1"
      >
        <figure>
          <img
            alt="Sarani"
            class="image"
            src="https://place-puppy.com/"
          />
          <figcaption
            class="linkText"
          >
            Sarani
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</div>
`;
